tests/publiccloud/aistack_basic.pm

```diff
@@ -0,0 +1,179 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Basic aistack test
+
+# Summary: This test performs the following actions
+#  - Create a VM in EC2 using SLE-Micro-BYOS later than 6.0 version
+#  - Install the required dependencies to install the aistack helm chart
+#  - Test access to OpenWebUI and run integration tests with Ollama and MilvusDB
+# Maintainer: Yogalakshmi Arunachalam <yarunachalam@suse.com>
+#
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use publiccloud::utils qw(is_byos registercloudguest register_openstack);
+use publiccloud::ssh_interactive 'select_host_console';
+use transactional;
+use containers::k8s;
+use strict;
+use warnings;
+use utils;
+use publiccloud::utils;
+use transactional qw(process_reboot trup_install trup_shell);
+use File::Basename;
+use version_utils;
+use Data::Dumper;
+
+sub tr_run_script {
+    my ($file_name, $ins_cmd) = @_;
+    # Write the script directly
+    script_run("echo '#!/bin/bash' > ~/$file_name");
+    script_run("echo 'set -e' >> ~/$file_name");
+    script_run("echo '$ins_cmd' >> ~/$file_name");
+    script_run("chmod 755 ~/$file_name");
+    script_run("cat ~/$file_name");
+    # Execute the script using trup_run
+    trup_call("run bash -c ~/$file_name");
+    record_info("Executed script $file_name.");
+    # Delete the script
+    script_run("rm ~/$file_name");
```

Comment:
Let's move all the trup-call *at end of common code* in the for-loop block and also `info` and `rm` conditioned:
`rm $file if (file exists)`

Reply:
fixed